### DISCUSSION TITLE
Prisma adds the same ALTER TABLE statement to all migrations

### DISCUSSION THREAD

**Discussion by viperet** [prisma/prisma#25638] [Q&A] [answered]
### Question

I have Screenshot table with url field:
```
CREATE TABLE `Screenshot` (
    `id` INTEGER NOT NULL AUTO_INCREMENT,
    `url` VARCHAR(191) NOT NULL
.....
    PRIMARY KEY (`id`)
) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
```
Schema:
```
model Screenshot {
  id                   Int          @id @default(autoincrement())
  url                  String
....
}
```
So DB is in sync with schema, varchar field 'url' is not nullable. But Prisma keeps adding the same SQL in every migration, and every time asks me to reset my DB.
```
ALTER TABLE `Screenshot` MODIFY `url` VARCHAR(191) NOT NULL;
```
![image](https://github.com/user-attachments/assets/d2e36e15-f047-4408-a0fa-d21df8076577)

This is a bit frustrating but I don't know how to approach this problem, how to debug it. It happens only on one of my projects, with only one field, so I don't know how to reproduce it.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @viperet 👋 

Can you try this command? 

```
npx prisma db pull --print
```
This will allow you to see what Prisma generates when it pulls from your database. Compare this with your current schema. There might be subtle syntax differences that are causing Prisma to think the schema has changed.

**Reply by viperet** [reply]
This command outputs the correct schema:
```
model Screenshot {
  id                   Int          @id @default(autoincrement())
  url                  String
...
}
```
The field is not nullable, but still each `prisma migrate dev` generates this migration SQL
```
/*
  Warnings:

  - Made the column `url` on table `Screenshot` required. This step will fail if there are existing NULL values in that column.

*/
-- AlterTable
ALTER TABLE `Screenshot` MODIFY `url` VARCHAR(191) NOT NULL;
```

**Reply by ludralph** [reply, marked as answer]
Hi @viperet 

Can you share a reproducible example so we can investigate this further?

**Reply by ludralph** [reply]
Hi @viperet 

It would be really helpful if you could provide the additional information that we asked about last time. If your problem is resolved by now or no longer relevant, you can simply close the Discussion as well.

**Reply by viperet** [reply]
Hi @ludralph, sorry for the delayed reply. Creating a reproducible example did help to resolve this. 
I found a migration SQL, prisma/migrations/RUN_LAST/migration.sql, created by one of the team members. He wanted to add a column comment, and since comments are not supported by Prisma, he did it using a migration. And that migration messed everything up.

**Reply by ludralph** [reply]
I'm glad to hear that you figured it out while trying to create a reproduction 😄.
